bin/oneoff/back_populate_gallery_activity

```diff
@@ -5,33 +5,46 @@
 # associating a GalleryActivity with a UserLevel rather than an Activity, as
 # part of deprecating the latter.
 
+# For performance reasons, we precompute the backfill data via the query
+#   SELECT
+#     gallery_activities.id,
+#     user_levels.id AS user_level_id,
+#     activities.level_source_id
+#   FROM (SELECT * FROM gallery_activities LIMIT 100000) gallery_activities
+#   INNER JOIN activities
+#     ON activities.id = gallery_activities.activity_id
+#   INNER JOIN user_levels
+#     ON user_levels.user_id = activities.user_id
+#       AND user_levels.level_id = activities.level_id;
+# saving the output to and reading the data from a CSV.
+
 require_relative '../../dashboard/config/environment'
+require 'csv'
 
-count = GalleryActivity.
-  where('user_level_id <> ? OR level_source_id <> ?', nil, nil).
-  count
-puts "UPDATING GalleryActivity: #{count} records"
-
-updates = 0
-GalleryActivity.
-  where('user_level_id <> ? OR level_source_id <> ?', nil, nil).
-  find_each(batch_size: 10_000) do |gallery_activity|
-  # We grab the user_level_id and level_source_id from data in the associated
-  # activity.
-  activity = gallery_activity.activity
-  user_level = UserLevel.find_by(user_id: activity.user_id, level_id: activity.level_id)
-
-  next unless user_level
-
-  gallery_activity.update!(
-    user_level_id: user_level.id,
-    level_source_id: activity.level_source_id
-  )
-
-  updates += 1
-  if updates % 10_000 == 0
-    puts "  UPDATED GalleryActivity: #{updates} records"
-  end
+TRANSACTION_SIZE = 5_000
+TIME_NOW = Time.now.freeze
+
+filename = ARGV[0]
+exit(-1) unless filename
+
+puts "READING #{filename}..."
+
+data = []
+CSV.foreach(filename) do |line|
+  gallery_activity_id = line[0]
+  user_level_id = line[1]
+  data << [gallery_activity_id, user_level_id, level_source_id]
```

Comment:
looks like you're missing a `level_source_id = line[2]`, although it's not clear to me why this can't just be `data = CSV.read(filename)`